core/src/main/java/org/lflang/generator/c/CWatchdogGenerator.java

```diff
@@ -83,8 +92,7 @@ protected static int generateInitializeWatchdogs(CodeBuilder code, ReactorInstan
     if (foundOne) {
       code.pr(temp.toString());
     }
-    code.pr("SUPPRESS_UNUSED_WARNING(_lf_watchdog_count);");
-    return watchdogCount;
+    enclaveInfo.numWatchdogs += watchdogCount;
```

Comment:
This doesn't look quite right.  Won't `watchdogCount` always be 0?

Reply:
I think you overlooked the problem the place where it is incremented. Right above